File: polyglot/seis_native/zig/seis_native_kernel.zig
pub const NativeLane = struct {
    name: []const u8,
    score: u8,
};

pub const lanes = [_]NativeLane{
    .{ .name = "Apple First", .score = 100 },
    .{ .name = "Data AI", .score = 88 },
    .{ .name = "Systems", .score = 84 },
    .{ .name = "Android", .score = 76 },
    .{ .name = "Windows", .score = 72 },
    .{ .name = "Infrastructure", .score = 70 },
};

pub fn topLane() NativeLane {
    return lanes[0];
}

File: polyglot/seis_native/cpp/seis_native_kernel.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

struct NativeRoadmapItem {
    std::string lane;
    int score;
    std::string language;
};

std::vector<NativeRoadmapItem> roadmap() {
    return {
        {"Apple First", 100, "Swift"},
        {"Data AI", 88, "Python"},
        {"Systems", 84, "Rust"},
        {"Android", 76, "Kotlin"},
        {"Windows", 72, "CSharp"},
        {"Infrastructure", 70, "Go"}
    };
}

NativeRoadmapItem top_lane() {
    auto items = roadmap();
    return *std::max_element(items.begin(), items.end(), [](const auto& left, const auto& right) {
        return left.score < right.score;
    });
}

int main() {
    const auto top = top_lane();
    std::cout << top.lane << " " << top.score << " " << top.language << "\n";
    return 0;
}
